crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -321,24 +390,38 @@ impl Metrics {
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error("failed to read data from database")]
+    #[error("failed communication with the database")]
     DbError(#[from] anyhow::Error),
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("failed communication with the database")]
+pub struct DatabaseError(#[from] pub sqlx::Error);
+
 pub mod error {
     use super::*;
 
     #[derive(Debug, thiserror::Error)]
     pub enum Auction {
-        #[error("failed to read data from database: {0}")]
-        DbError(#[source] anyhow::Error),
+        #[error("failed communication with the database: {0}")]
+        DatabaseError(#[from] sqlx::Error),
         #[error("auction data not found in the database")]
         Missing,
         #[error("failed dto conversion from database: {0} for order: {1}")]
         FeePolicy(dto::fee_policy::Error, domain::OrderUid),
-        #[error(transparent)]
-        Price(#[from] domain::auction::InvalidPrice),
-        #[error("jit order owners not found for an existing auction id")]
-        MissingJitOrderOwners,
+        #[error("failed to fetch price for token: {0}")]
+        Price(eth::TokenAddress),
+    }
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Winner {
```

Comment:
These errors are not specific to a `Winner` but to loading any solution from the Database. Can we please make it solution generic?

Reply:
resolved, if you meant this needs to be renamed